### ISSUE TITLE
Roadmap for Browser Support in 2.0?

### ISSUE THREAD

**Issue by samypr100** [image-size/image-size#384]
Since I'm seeing a sneak peek at what might be 2.0 and native ESM in #370, I'm curious if there's any thoughts in the roadmap for a browser bundle version of the library.

This is such an amazing library and there's been a lot of improvements on the build tooling, standards, and browsers that might be worth asking again since the state of things back then in #62. I'm also interested partly on this since I'm trying to avoid using `new Image()` as I only need image dimensions and your library is way way faster.

I'm thinking a `dist` bundle that has both something like `index.browser.js` vs `index.js` where the `index.browser.js` has limited features/formats (e.g. no fs support at least not until [FS Browser API](https://wicg.github.io/file-system-access/) maybe becomes a thing).

Thoughts?

**Comment by samypr100** [comment]
Closing as 2.0 has initial browser support now 🎉

**Comment by netroy** [comment]
I haven't published these updates, as I'm still sorting out some ESM stuff. but hopefully this will be out as `2.0.0-alpha.2` over the next days.
I'll let you folks know.
Thanks for taking a lead on this 🙏🏽

**Comment by samypr100** [comment]
Awesome, can't wait to try `2.0.0-alpha.2` it via npm 🚀

**Comment by johnthagen** [comment]
@netroy We'd also love to try out the `alpha.2` when it's published to NPM. Thanks!

**Comment by mcdenhoed** [comment]
Hey @netroy, any ETA on publishing `2.0.0-alpha.2`?

**Comment by mcdenhoed** [comment]
Hey @netroy! Thanks for publishing! It looks like some of the updates following the closing of this MR broke browser support in some cases.

If I make a new empty angular project and use `imageSize` from it as follows:

```typescript
import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterOutlet } from '@angular/router';
import { imageSize } from "image-size";

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [CommonModule, RouterOutlet],
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'ng-image-size-test';
  constructor() {
    imageSize(new Uint8Array())
  }
}

```

I get an error like this:

```
✘ [ERROR] Could not resolve "./types"

    node_modules/image-size/dist/esm/index.mjs:4:22:
      4 │ var types_1 = require("./types");
        ╵                       ~~~~~~~~~

✘ [ERROR] Could not resolve "./lookup"

    node_modules/image-size/dist/esm/index.mjs:6:23:
      6 │ var lookup_1 = require("./lookup");
        ╵                        ~~~~~~~~~~

```

Or sometimes with different typescript settings I get something more like this (thrown from the file `image-ops.ts` in my actual project):

```
./src/image-ops.ts:19:16-25 - Error: export 'imageSize' (imported as 'imageSize') was not found in 'image-size' (module has no exports)

```

I suspect this change might be responsible: https://github.com/image-size/image-size/commit/5bc227862be7aaeac80b94b41685d18b0974e452#diff-d5a4b3a0309f2337144861084c1014523cced561eb722cb9684c63f5e41e0567R1


```typescript
export { lookup as imageSize, disableTypes, types } from './lookup'
```

I was using a vendored copy of d40cdc06e2921cde579dca98a67f05983feac6a6 which worked great, and something broke in-between that commit and the version released on npm as `2.0.0-alpha.2`, which I suspect was 0d81314abbff8bf8f884d04e5ea730df30b6a173

**Comment by tmadeira** [comment]
@mcdenhoed, have you found any workaround for that issue? I'm also getting:

```
Error: export 'imageSize' (imported as 'imageSize') was not found in 'image-size' (module has no exports)
```

using `2.0.0-beta.1`.

**Comment by netroy** [comment]
Unfortunately there is no workaround, as `2.0.0-beta.1` isn't built correctly to support this.
Since I don't use ESM yet, and don't work on any frontend code, I don't really know what exactly to fix. If someone can help fix the build so that it works without issues on CJS and ESM, we can finally release a 2.0.

**Comment by mcdenhoed** [comment]
I might have a try at reverting this change:

https://github.com/image-size/image-size/commit/5bc227862be7aaeac80b94b41685d18b0974e452#diff-d5a4b3a0309f2337144861084c1014523cced561eb722cb9684c63f5e41e0567R1

I think this is the change that broke browser support after it was added in https://github.com/image-size/image-size/commit/d40cdc06e2921cde579dca98a67f05983feac6a6

**Comment by netroy** [comment]
Published #399 as `2.0.0-beta.2`. Can some of you folks try upgrading to that package and let me know if it works as expected for you?
